### ISSUE TITLE
Configuration option: Don't skip collapsed cells when selecting

### ISSUE THREAD

**Issue by lukelbd** [jupyterlab/jupyterlab#11162]
### Problem

#6356 changes the default jupyterlab behavior to skip collapsed cells when scrolling with the arrows or `j` + `k`. Would it be possible to add a configuration option to switch back to the old behavior?

I personally dislike this new behavior. I often know which code is in a collapsed cell and need to re-execute it, but prefer to keep it collapsed because it is unusually long. For example: I have a >100 line function definition, and subsequent cells only ~5-10 lines long that call the function with different arguments.

Previously, to re-execute a collapsed cell block, I would simply navigate to it with the arrows or `j` + `k`. Now, I have to select the collapsed cell with the mouse pointer, or `Shift + Enter` through adjacent cells until I reach the cell I want to execute (which is not always practical/possible).

### Proposed Solution

Add a configuration option to include collapsed cells when scrolling with the arrows or `j` + `k`.

**Comment by tonyfast** [comment]
i'm running into similar issues. at some point, we could navigate to any cell with <kdb>Up + Down</kbd> keys or <kdb>J + K </kbd>. 

currently, collapsed cells are omitted from the navigation. they cannot be accessed with a keyboard and require a mouse. when all cells are collapsed the navigation defaults to the first cell and we cannot move around with the keyboard.

**Comment by tonyfast** [comment]
after some thought i'm add the :bug: label because when cells are collapsed there is no way to navigate to them with a keyboard, or i haven't found one yet.

**Comment by lukelbd** [comment]
Encountered this again in the latest project so just wanted to bump this issue @tonyfast. Could this be a relatively simple fix?

**Comment by lukelbd** [comment]
I think this can be implemented by changing the below `selectAbove()` and `selectBelow()` lines from:

```ts
if (!possibleNextCell.inputHidden && !possibleNextCell.isHidden) { 
   break; 
 }
```

to the following:

```ts
 if  (!possibleNextCell.isHidden) { 
   break; 
 } 
```

I'll try a pull request.

https://github.com/jupyterlab/jupyterlab/blob/48691c6a3040af46c4ed084a06c2d83c2d9e9236/packages/notebook/src/actions.tsx#L924-L926
https://github.com/jupyterlab/jupyterlab/blob/48691c6a3040af46c4ed084a06c2d83c2d9e9236/packages/notebook/src/actions.tsx#L972-L974

**Comment by krassowski** [comment]
Cross-posting my questions from review of https://github.com/jupyterlab/jupyterlab/pull/16881
- should it be possible to press `<Right>` to expand the collapsed cells?
- should we have two variants of `<Up>`, `<Down>`, `k`, `j`? Like `Alt` + `Up` expands, just `Up` does not (or vice versa)?

**Comment by tonyfast** [comment]
> * should it be possible to press `<Right>` to expand the collapsed cells?

yes. the action needs to be [reversible for cognitive accessibility](https://www.w3.org/WAI/WCAG2/supplemental/patterns/o4p02-back-undo/)
 
> * should we have two variants of `<Up>`, `<Down>`, `k`, `j`? Like `Alt` + `Up` expands, just `Up` does not (or vice versa)?

no

both of the answers to this question are in the spec for a [`role=tree`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tree_role#keyboard_interactions) which is basically a collapsible list with recommended keyboard patterns.

**Comment by ledovsky** [comment]
Hi! Found an unfinished PR for this issue. I also suffer from the described problem.
Don't you mind if I take the issue and try fix it?
